tests/utils.go

```diff
@@ -469,6 +469,64 @@ func GetSupportedCPUModels(nodes k8sv1.NodeList) []string {
 	return cpus
 }
 
+func GetSupportedCPUFeaturesAmongAllTheNodes(nodes k8sv1.NodeList) []string {
+	var featureDenyList = map[string]bool{
+		"svm": true,
+	}
+	var supportedFeatures []string
+	for label := range nodes.Items[0].Labels {
+		if strings.Contains(label, services.NFD_CPU_FEATURE_PREFIX) {
+			feature := strings.TrimPrefix(label, services.NFD_CPU_FEATURE_PREFIX)
+			if _, ok := featureDenyList[feature]; !ok {
+				supportedFeatures = append(supportedFeatures, feature)
+			}
+		}
+	}
+
+	for _, node := range nodes.Items {
+		var currFeatures []string
+		for key := range node.Labels {
+			if strings.Contains(key, services.NFD_CPU_FEATURE_PREFIX) {
+				feature := strings.TrimPrefix(key, services.NFD_CPU_FEATURE_PREFIX)
+				if _, ok := featureDenyList[feature]; !ok {
+					currFeatures = append(currFeatures, feature)
+				}
+			}
+		}
+		supportedFeatures = setIntersection(supportedFeatures, currFeatures)
+	}
+
+	return supportedFeatures
+}
+func setIntersection(firstSet, secondSet []string) (intersection []string) {
+	firstSetBoolMap := make(map[string]bool)
+	var setOfFeaturesWithDups []string
+
+	for _, element := range firstSet {
+		firstSetBoolMap[element] = true
+
+	}
+
+	for _, element := range secondSet {
+		if firstSetBoolMap[element] {
+			setOfFeaturesWithDups = append(setOfFeaturesWithDups, element)
+		}
+	}
+	intersection = removeDups(setOfFeaturesWithDups)
+	return intersection
+}
+
+func removeDups(elements []string) (intersection []string) {
```

Comment:
Named return values are imo very hard to follow.

Can't you just return the slice ?

Reply:
i see you point but this function is very smalland i don't see much difference between declaring the return var in the func's signature or in the first line of the func. :)

Reply:
fair enough.